File: JavaScript/08. Functions/Functions-Homework/Functions-Homework/06. LargerThanNeighbours/LargerThanNeighbours.js
/*
 * Problem 6. Larger than neighbours

Write a function that checks if the element at given position in given array of integers is bigger than its two neighbours (when such exist).

 */

function LargerThanNeighbours(arr, index) {
    arr = arr || [];
    index = index || 0;

    var result = '';

    if (index > 0 && index < arr.length - 1) {
        if (arr[index - 1] < arr[index] && arr[index] > arr[index + 1]) {
            result = "The element " + arr[index] + " is bigger than its two neighbours -> (" + arr.slice(index - 1, index + 2) + ")";
        } else {
            result = "The element " + arr[index] + " is NOT bigger than its two neighbours -> (" + arr.slice(index - 1, index + 2) + ")";
        }

    } else {
        result = 'Neighbour elements does not exist!';
    }
    return result;
}

var arr = [1, 2, 3, 2, 1],
    index1 = 2,
    index2 = 3;

console.log("Test1 index = " + index1);
console.log(LargerThanNeighbours(arr, index1));
console.log("Test2 index = " + index2);
console.log(LargerThanNeighbours(arr, index2));

File: C# II/03. Methods/Methods-Homework/05. LargerThanNeighbours/LargerThanNeighbours.cs
using System;

public class LargerThanNeighbours
{
    /*
      5. Larger than neighbours

        Write a method that checks if the element at given position 
        in given array of integers is larger than its two neighbours 
        (when such exist).

    */

    public static bool LargerThanNeighboursMethod(int[] arr, int position)
    {
        if((position > 0 && position < arr.Length -1) && (arr[position-1] < arr[position])
            && (arr[position] > arr[position + 1]))
            return true;
        return false;
    }

    static void Main(string[] args)
    {
        Console.Write("Enter integers separated by space: \n");
        int[] arr = Array.ConvertAll(Console.ReadLine().Split(), int.Parse);
        Console.Write("Enter a position: ");
        int position = int.Parse(Console.ReadLine());
        if (position > 0 && position < arr.Length)
        {
            Console.Write("The element {0} on position {1} is ", arr[position],
                position);
            if(LargerThanNeighboursMethod(arr, position))
            {
                Console.WriteLine("larger than it's neighbours.");
            }
            else
            {
                Console.WriteLine(" NOT larger than it's neighbours.");
            }
        }
        else
        {
            Console.WriteLine("The index is out range!");
        }
    }
}

